### PR TITLE
varlink: fix more/continues method calls, and correctly apply method call timeout

### PR DESCRIPTION
**Pull request by poettering**
And three minor cleanups

### CODE REVIEW

**Comment by poettering** [comment]
(this are bugfix patches split out of the homed PR #14096, since they probably should be applied indepdendently and earlier, given that journald already uses varlink)

File: src/shared/varlink.c [hunk 1]
```diff
@@ -715,14 +715,19 @@ static int varlink_dispatch_reply(Varlink *v) {
                         goto invalid;
         }
 
+        /* Replies with 'continue' set are only OK we set 'more' when we initiated the method */
```
Comment (keszybz) [inline src/shared/varlink.c]: Something's wrong with this sentence.

File: src/shared/varlink.h [hunk 2]
```diff
@@ -86,6 +86,10 @@ int varlink_callb(Varlink *v, const char *method, JsonVariant **ret_parameters,
 int varlink_invoke(Varlink *v, const char *method, JsonVariant *parameters);
 int varlink_invokeb(Varlink *v, const char *method, ...);
 
+/* Enqueue method call, expect a reply now, and possibly more later, which are all delivered to the reply callback */
+int varlink_observe(Varlink *v, const char *method, JsonVariant *parameters);
+int varlink_observeb(Varlink *v, const char *method, ...);
```
Comment (keszybz) [inline src/shared/varlink.h:91]: Those functions are not used anywhere in this PR, right?

File: src/shared/varlink.c (hunk 1 above)
Comment (poettering) [reply src/shared/varlink.c]: fixed

File: src/shared/varlink.h (hunk 2 above)
Comment (poettering) [reply src/shared/varlink.h:91]: nope, they are not. homed makes use of them though.

I was pondering whether to post this split out or not. The reason I did it after all was that the current code claimed it supported method calls with "more" set (i.e. method calls that have multiple replies, in a "subscribe" fashion), but actually was totally broken in this regard. Now, we don't use "more" yet in mainline, but I couldn't live with the fact that the code claimed to do something it didn't...

(there are also a number of other functions in varlink.c currently not used in mainline)

**Comment by poettering** [comment]
so, yeah i can split out that one commit or split it (so that the enum states are kept in, but the functions to enter them are not). But this appeared weird to me, since the code is thus "incomplete"... We could also remove the commit altogether, just ignoring the fact that the current code claims to support "more" but actually does not... what would you prefer?

**Comment by poettering** [comment]
i force pushed a new version that fixes that comment, but leaves everything else the way it is

File: src/shared/varlink.h (hunk 2 above)
Comment (keszybz) [reply src/shared/varlink.h:91]: OK, let's leave this in then. No need to make complicate things too much.

**Comment by keszybz** [comment]
CI passed previously, and now there's just a changed comment, so we don't need to test again. Let's merge.

**Pull request merged**